packages/analytics-js-integrations/src/integrations/XPixel/utils.js

```diff
@@ -0,0 +1,45 @@
+import { trackPropertyMapping } from '@rudderstack/analytics-js-common/constants/integrations/XPixel/constants';
+import { removeUndefinedAndNullValues } from '../../utils/commonUtils';
+import { constructPayload } from '../../utils/utils';
+
+const getContents = message => {
+  let contents = [];
+  const { properties } = message;
+  const { products } = properties;
+  if (Array.isArray(products)) {
+    contents = products.map(product => {
+      const singleProduct = {
+        content_type: product.contentType || product.content_type || 'product',
+        content_id: product.product_id,
+        content_name: product.name,
+        price: product.price,
+        num_items: product.quantity,
+        group_id: product.groupId || product.group_id,
+      };
+      return removeUndefinedAndNullValues(singleProduct);
+    });
+  }
```

Comment:
```suggestion
  }else {
  return undefined
  }
```

Reply:
This is done to not send empty contents array if we don't have any

Reply:
done